### PR TITLE
C2 amr a raouf c2 s05 facebook inviting

### PR DESCRIPTION
**Pull request by AmrARaouf**
ISSUE #184 

@Omar-Hossam 
@Mirna-Yacout 
@karimngr 
@SalmaFG 

Files to check:

/views/games/game.html.erb

/controllers/games_controller.rb

/models/gamers.rb

### CODE REVIEW

File: arability/app/models/gamer.rb [hunk 1]
```diff
@@ -1,28 +1,224 @@
...
+  has_one :authentication
+  has_and_belongs_to_many :prizes
+  has_and_belongs_to_many :trophies
+
   # Include default devise modules. Others available are:
   # :token_authenticatable, :encryptable, :confirmable, :lockable, :timeoutable and :omniauthable
+
   devise :database_authenticatable, :registerable,
-         :recoverable, :rememberable, :trackable, :validatable
+         :recoverable, :rememberable, :trackable, :validatable, :omniauthable, :omniauth_providers => [:facebook]
```
Comment (omar-mohyeldin) [inline arability/app/models/gamer.rb]: divide your line to be a maximum of 80 characters

File: arability/app/models/gamer.rb [hunk 2]
```diff
@@ -1,28 +1,224 @@
...
 
   # Setup accessible (or protected) attributes for your model
-  attr_accessible :email, :password, :password_confirmation, :remember_me, 
-                  :username, :country, :education_level, :date_of_birth
-                  
+  attr_accessible :email, :password, :password_confirmation, :remember_me, :gender,
```
Comment (omar-mohyeldin) [inline arability/app/models/gamer.rb]: divide your line to be a maximum of 80 characters

File: arability/app/controllers/games_controller.rb [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+  #   The user is not signed in: S/He'll be redirected to the sign in page
+  def post
+    if current_gamer != nil
+      if !current_gamer.is_connected_to_facebook 
+        redirect_to "/gamers/edit", flash: {notice: t(:connect_your_account)}
+      else
+        begin
+          token = current_gamer.get_token
+          @graph = Koala::Facebook::API.new(token)
+          @graph.put_wall_post("Checkout the new Arability game @ www.arability.net\n" + Time.now.to_s)
```
Comment (omar-mohyeldin) [inline arability/app/controllers/games_controller.rb]: divide line to have a maximum of 80 characters

**Comment by AmrARaouf** [comment]
@Omar-Hossam Done. Please review again.

File: arability/app/controllers/games_controller.rb [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+          redirect_to "/game", flash: {success: t(:shared_on_fb)}
+        rescue Koala::Facebook::AuthenticationError
```
Comment (omar-mohyeldin) [inline arability/app/controllers/games_controller.rb:75]: indent line to be alligned with rest of code

File: arability/app/controllers/games_controller.rb [hunk 5]
```diff
@@ -0,0 +1,78 @@
...
+          redirect_to "/gamers/auth/facebook"
+        rescue Koala::Facebook::ClientError
```
Comment (omar-mohyeldin) [inline arability/app/controllers/games_controller.rb:77]: indent line to be alligned with rest of code

File: arability/app/views/games/game.html.erb [hunk 6]
```diff
@@ -0,0 +1,202 @@
+<script type="text/javascript" charset="utf-8">
+
```
Comment (omar-mohyeldin) [inline arability/app/views/games/game.html.erb]: remove spaces from empty line

**Comment by omar-mohyeldin** [comment]
@AmrARaouf modify your code

**Comment by ghost** [comment]
@AmrARaouf  scenario reviewed

**Comment by AmrARaouf** [comment]
@Omar-Hossam check again please

**Comment by omar-mohyeldin** [comment]
@AmrARaouf code reviewed

File: arability/app/models/gamer.rb [hunk 7]
```diff
@@ -1,28 +1,272 @@
...
+  # Failure:
+  #   None
+  def is_connected_to_facebook
+    token != nil
+  end
+
+  # Author:
+  #   Amr Abdelraouf
+  # Description:
+  #   Return's the user's facebook access token
```
Comment (salmafg) [inline arability/app/models/gamer.rb]: Returns*

File: arability/app/models/gamer.rb [hunk 8]
```diff
@@ -1,28 +1,272 @@
...
+  # Params:
+  #   None
+  # Success:
+  #   Return's the user's access token
```
Comment (salmafg) [inline arability/app/models/gamer.rb]: Returns*

**Comment by Mirna-Yacout** [comment]
UML reviewed

**Pull request merged**